modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4883,10 +4883,10 @@ class CV_EXPORTS LineIterator
     LineIterator( Point pt1, Point pt2,
                   int connectivity = 8, bool leftToRight = false )
     {
-        init(0, Rect(std::min(pt1.x, pt2.x),
-                     std::min(pt1.y, pt2.y),
-                     std::max(pt1.x, pt2.x) - std::min(pt1.x, pt2.x) + 1,
-                     std::max(pt1.y, pt2.y) - std::min(pt1.y, pt2.y) + 1),
+        init(0, Rect((std::min)(pt1.x, pt2.x),
+                     (std::min)(pt1.y, pt2.y),
+                     (std::max)(pt1.x, pt2.x) - (std::min)(pt1.x, pt2.x) + 1,
+                     (std::max)(pt1.y, pt2.y) - (std::min)(pt1.y, pt2.y) + 1),
```

Comment:
Need to revert this.

Include `imgproc.hpp` before openexr headers.

Reply:
The problem is that all sorts of headers may `#define` macros `min` and `max`. In my case, it was [`JXRGlue.h`](https://github.com/4creators/jxrlib/blob/cae40c1367375fadd86c5bf16c18ad04e8e268b8/jxrgluelib/JXRGlue.h#L42-L47). While this header may not have anything to do with OpenCV (or OpenEXR), I believe this change may help also other users (including people who may `#include <Windows.h>`).

I've personally found [this solution](https://stackoverflow.com/a/13420838/19254) to generally work best.

With all this said, if you still think this change needs to be reverted, I'll do it. Just wanted to make clear why I introduced it in the first place.

Reply:
Please

> Include `imgproc.hpp` before openexr headers.

This should resolve this problem.

Reply:
As I tried to say, this problem isn't related to OpenEXR (but to [jxrlib](https://github.com/4creators/jxrlib) instead). But ok, have now reverted this anyway.